lib/phoenix/controller.ex

```diff
@@ -474,12 +474,14 @@ defmodule Phoenix.Controller do
     end)
   end
 
+  @type view :: {format :: atom, view :: atom}
```

Comment:
This makes the type deeply recursive, which is not what we want. The format is also only supported as a keyword list.